packages/components/alert/src/Alert.tsx

```diff
@@ -0,0 +1,187 @@
+/**
+ * @license
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
+ */
+
+import { Ref, Transition, computed, defineComponent, normalizeClass, ref } from 'vue'
+
+import { callEmit } from '@idux/cdk/utils'
+import { useGlobalConfig } from '@idux/components/config'
+import { IxIcon } from '@idux/components/icon'
+
+import { AlertProps, alertIconMap, alertProps } from './types'
+
+export default defineComponent({
+  name: 'IxAlert',
+  inheritAttrs: false,
+  props: alertProps,
+  setup(props, { slots }) {
+    const isVisible = ref(true)
+    const hasDescription = ref(false)
+    const alertConfig = useGlobalConfig('alert')
+    const isShowIcon = computed(() => props.icon !== '')
+    const iconName = computed(() => props.icon ?? (alertConfig.icon || alertIconMap[props.type]))
+
+    const {
+      setPageTotal,
+      pageIndex,
+      pageText,
+      isShowPagination,
+      isLeftDisabled,
+      isRightDisabled,
+      handlePageLeft,
+      handlePageRight,
+    } = usePagination()
+    const {
+      alertClass,
+      iconClass,
+      contentClass,
+      closeIconClass,
+      paginationClass,
+      pageTextClass,
+      leftIconClass,
+      rightIconClass,
+    } = useClass(props, hasDescription, isLeftDisabled, isRightDisabled)
+    const { mergedClosable, handleCloseClick } = useClose(props, alertConfig.closable, isVisible)
+
+    return () => {
+      const children = slots.default?.() ?? []
+      setPageTotal(children.length)
```

Comment:
```suggestion
  const chilren = computed(()=> slots.default?.() || NoopArray)
  const pageTotal = computed(() => chilren.value.length)
```
这里可以改成这样， render 函数外面用 computed 缓存起来

Reply:
改成了`const children = computed(() => flattenNode(slots.default?.() ?? []))`